From be6c23f9fc6863944cdb3abb3c8b2672eda256c1 Mon Sep 17 00:00:00 2001
From: Chris Renshaw <osm0sis@outlook.com>
Date: Sun, 18 Feb 2018 07:54:58 -0400
Subject: [PATCH 14/34] fix fsck.minix, mkfs.minix: undef HAVE_SETBIT,
 MINIX2_SUPER_MAGIC, MINIX2_SUPER_MAGIC

"there is no setbit/clrbit in bionic"
"MINIX2_SUPER_MAGIC / MINIX2_SUPER_MAGIC2 defined in sys/vfs.h, undefine
it to use it in the enum"

Patch by Tias Guns <tias@ulyssis.org>, based on 'no-setbit' and
'undefine-minix2-magic-to-use-in-enum' by Dan Drown
http://dan.drown.org/android/src/busybox/
Rebased for busybox 1.28.1 by Chris Renshaw <osm0sis@outlook.com>
---
 include/platform.h | 1 +
 util-linux/minix.h | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/include/platform.h b/include/platform.h
index 9e1fb047d..668b50df4 100644
--- a/include/platform.h
+++ b/include/platform.h
@@ -548,6 +548,7 @@ typedef unsigned smalluint;
 # undef HAVE_UNLOCKED_LINE_OPS
 # undef HAVE_NET_ETHERNET_H
 # undef HAVE_PRINTF_PERCENTM
+# undef HAVE_SETBIT
 #endif
 
 /*
diff --git a/util-linux/minix.h b/util-linux/minix.h
index 83ffe6da5..91ced691c 100644
--- a/util-linux/minix.h
+++ b/util-linux/minix.h
@@ -54,6 +54,8 @@ struct minix_dir_entry {
 /* Believe it or not, but mount.h has this one #defined */
 #undef BLOCK_SIZE
 
+#undef MINIX2_SUPER_MAGIC
+#undef MINIX2_SUPER_MAGIC2
 enum {
 	BLOCK_SIZE              = 1024,
 	BITS_PER_BLOCK          = BLOCK_SIZE << 3,
-- 
2.31.1

